r2-navigator/src/main/java/org/readium/r2/navigator/epub/EpubNavigatorFragment.kt

```diff
@@ -0,0 +1,483 @@
+                    // TODO based on device
+                    // TODO decide if 1 page or 2 page
+                    setCurrent(resourcesSingle)
+                }
+            }
+        }
+
+        return true
+    }
+
+    override fun go(link: Link, animated: Boolean, completion: () -> Unit): Boolean {
+        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
+    }
+
+    override fun onProgressionChanged(progression: Double) {
+        notifyCurrentLocation()
+    }
+
+    override fun goForward(animated: Boolean, completion: () -> Unit): Boolean {
+        launch {
+            if (resourcePager.currentItem < resourcePager.adapter!!.count - 1) {
+
+                resourcePager.setCurrentItem(resourcePager.currentItem + 1, animated)
+
+                if (currentFragment?.activity?.layoutDirectionIsRTL() ?: publication.contentLayout.readingProgression == ReadingProgression.RTL) {
+                    // The view has RTL layout
+                    currentFragment?.webView?.apply {
+                        progression = 1.0
+                        setCurrentItem(numPages - 1, false)
+                    }
+                } else {
+                    // The view has LTR layout
+                    currentFragment?.webView?.apply {
+                        progression = 0.0
+                        setCurrentItem(0, false)
+                    }
+                }
+            }
+        }
+        return true
+    }
+
+    override fun goBackward(animated: Boolean, completion: () -> Unit): Boolean {
+        launch {
+            if (resourcePager.currentItem > 0) {
+
+                resourcePager.setCurrentItem(resourcePager.currentItem - 1, animated)
+
+                if (currentFragment?.activity?.layoutDirectionIsRTL() ?: publication.contentLayout.readingProgression == ReadingProgression.RTL) {
+                    // The view has RTL layout
+                    currentFragment?.webView?.apply {
+                        progression = 0.0
+                        setCurrentItem(0, false)
+                    }
+                } else {
+                    // The view has LTR layout
+                    currentFragment?.webView?.apply {
+                        progression = 1.0
+                        setCurrentItem(numPages - 1, false)
+                    }
+                }
+            }
+        }
+        return true
+    }
+
+    private val r2PagerAdapter: R2PagerAdapter
+        get() = resourcePager.adapter as R2PagerAdapter
+
+    private val currentFragment: R2EpubPageFragment? get() =
+        r2PagerAdapter.mFragments.get(r2PagerAdapter.getItemId(resourcePager.currentItem)) as? R2EpubPageFragment
+
+    override fun onTap(point: PointF): Boolean {
+        (this.listener as Navigator.VisualListener).onTap(point)
+        return super.onTap(point)
+    }
+
+    override val readingProgression: ReadingProgression
+        get() = TODO("Not yet implemented")
```

Comment:
I don't know how and where to set this variable.

Reply:
You just need to report the one from the `Publication`, this is an alias to expose it to reading apps.

```suggestion
        get() = publication.contentLayout.readingProgression
```